src/Accounts/Accounts/Utilities/CommandNotFoundHelper.cs

```diff
@@ -0,0 +1,298 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Reflection;
+using System.Text;
+using System.Text.RegularExpressions;
+using FuzzySharp.SimilarityRatio;
+using FuzzySharp.SimilarityRatio.Scorer.StrategySensitive;
+using Microsoft.Azure.Commands.Profile.Properties;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Commands.Profile.Utilities
+{
+    /// <summary>
+    /// Helper class to provide suggestions when the command is not found.
+    /// </summary>
+    public static class CommandNotFoundHelper
+    {
+        public static bool EnableFuzzyString = true;
+        private const string dataResourceName = "Microsoft.Azure.Commands.Profile.Utilities.CommandMappings.json";
+
+        private static CommandInvocationIntrinsics Cii = null;
+        private static readonly Regex AzOrAzureRMRegex = new Regex(@"^\w+-az\w+$", RegexOptions.IgnoreCase);
+        private static readonly Regex AzureRMRegex = new Regex(@"^\w+-azurerm\w+$", RegexOptions.IgnoreCase);
+        private static readonly Lazy<AllCommandInfo> LazyAllCommandInfo = new Lazy<AllCommandInfo>(LoadAllCommandInfo);
+
+        /// <summary>
+        /// Mappings from command names to their module names.
+        /// </summary>
+        private static readonly Lazy<IDictionary<string, string>> LazyCommandToModuleMappings = new Lazy<IDictionary<string, string>>(() =>
+        {
+            var moduleToCommandMappings = LazyAllCommandInfo.Value.Modules;
+            var mappings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var moduleName in moduleToCommandMappings.Keys)
+            {
+                foreach (var commandName in moduleToCommandMappings[moduleName].Keys)
+                {
+                    mappings.Add(commandName, moduleName);
+                }
+            }
+            return mappings;
+        });
+
+        /// <summary>
+        /// Mappings from command names to their migration guides.
+        /// </summary>
+        private static readonly Lazy<IDictionary<string, MemoryMigrationDetails>> LazyCommandToMigrationMappings
+            = new Lazy<IDictionary<string, MemoryMigrationDetails>>(() =>
+            {
+                var mappings = new Dictionary<string, MemoryMigrationDetails>(StringComparer.OrdinalIgnoreCase);
+                foreach (var pair in LazyAllCommandInfo.Value.Migration)
+                {
+                    foreach (var migration in pair.Value)
+                    {
+                        mappings.Add(migration.Key,
```

Comment:
For robustness，
```suggestion
                        if(!mappings.ContainsKey(migration.Key))mappings.Add(migration.Key,
```
Otherwise, error will throw if key exists
(No sure whether the syntax is correct

Reply:
That's a very good idea. I'll use `mappings[migration.Key] = ...` for simplicity.